### DISCUSSION TITLE
Plot from slice points

### DISCUSSION THREAD

**Discussion by Dementiy** [pyvista/pyvista#2798] [Q&A] [answered]
Hello.

I created a slice using `slice_along_axis`:

```python
sl = mesh.slice_along_aixs(line)
```

And then I want to plot this slice using matplotlib:

```python
# ...
points = np.asarray(sl.points)
X, Y, Z = points.T
ax.plot(X, Y, Z)
```
![image](https://user-images.githubusercontent.com/4396325/174115101-1f4e7ddc-52d7-4710-a16e-97b1f4d45c1d.png)

Order of points is incorrect. When I plot using pyvista everything looks good:

```python
p = pv.Plotter()
p.add_mesh(sl, color="r")
p.show()
```

![image](https://user-images.githubusercontent.com/4396325/174115767-3b4590af-5ac2-46f2-a7bd-87442ece1448.png)

How to get the right order of points from slice?

**Comment by whophil** [comment, marked as answer]
Hi @Dementiy 

As you've found, the order of the points resulting from `slice` are not ordered the same as their connectivity.

In order to get the correct connectivity, you can use the `strip` filter.

A complete example follows:

```python
import pyvista as pv
import matplotlib.pyplot as plt

def _extract_lines_info(lines):
    """Extract a list of line definitions from `PolyData.lines`"""
    offset = 0
    line_conn = []
    while offset < len(lines):
        line_length = lines[offset]
        line_conn.append(tuple(lines[offset + 1 : offset + 1 + line_length]))
        offset += line_length + 1

    return line_conn

# load mesh
mesh = pv.Sphere()

# slice at x = 0
res = mesh.slice('x', origin=(0, 0, 0))

# use .strip filter to join into lines
res = res.strip()
lines = _extract_lines_info(res.lines)

# plot the points directly
fig, axs = plt.subplots(2, 1, sharex=True, sharey=True, figsize=(12, 8))

y = res.points[:, 1]
z = res.points[:, 2]
axs[0].plot(y, z, '-o')
axs[0].set_title('points from slice')

# plot the lines
for i, line_indices in enumerate(lines):
    y = res.points[line_indices, 1]
    z = res.points[line_indices, 2]
    axs[1].plot(y, z, '-o', label=f'line {i}')
axs[1].set_title('ordered points after .strip')
plt.legend()
plt.show()
```

<img width="466" alt="image" src="https://user-images.githubusercontent.com/2186528/174667883-a9f89f18-100a-4345-9489-3874565f97c1.png">

**Reply by Dementiy** [reply]
Hi, @whophil , it works perfectly, thanks.
